### PR TITLE
[Performance] Improve show proc statistic #6477 

### LINKED ISSUE #6477: [Performance] show proc statistic takes too long
```sql
show proc '/statistic';
```

This query will calculate all replicas, which will takes very long on a large cluster.
Statistic don't need to calculate every time, one update in one minute is enough.
This could calculate in parallel.

### PR DESCRIPTION
**Pull request by ccoffline**
## Proposed changes

To speed up `show proc '/statistic'`, this pr refactor the statistic structure to make the logic clear, and use parallel computing.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)
- [x] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

- [x] I have created an issue on (Fix #6477) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

**Comment by ccoffline** [comment]
There is a problem with the use of `parallelStream` and it's `ForkJoinPool`.

**Comment by ccoffline** [comment]
> There is a problem with the use of `parallelStream` and it's `ForkJoinPool`.

avoid to use nested `parallelStream`, problem solved

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/StatisticProcDir.java [hunk 1]
```diff
@@ -58,86 +60,95 @@
...
+            this.dbNum = 1;
+            this.unhealthyTabletIds = new HashSet<>();
+            this.inconsistentTabletIds = new HashSet<>();
+            this.unrecoverableTabletIds = new HashSet<>();
+            this.cloningTabletIds = AgentTaskQueue.getTask(db.getId(), TTaskType.CLONE)
+                    .stream().map(AgentTask::getTabletId).collect(Collectors.toSet());
+
+            SystemInfoService infoService = Catalog.getCurrentSystemInfo();
             List<Long> aliveBeIdsInCluster = infoService.getClusterBackendIds(db.getClusterName(), true);
             db.readLock();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/StatisticProcDir.java:137]: I think db lock is not needed here.

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/StatisticProcDir.java [hunk 2]
```diff
@@ -58,86 +60,95 @@
...
                             ReplicaAllocation replicaAlloc = olapTable.getPartitionInfo().getReplicaAllocation(partition.getId());
-                            ++dbPartitionNum;
+                            ++partitionNum;
                             for (MaterializedIndex materializedIndex : partition.getMaterializedIndices(IndexExtState.VISIBLE)) {
-                                ++dbIndexNum;
+                                ++indexNum;
                                 for (Tablet tablet : materializedIndex.getTablets()) {
-                                    ++dbTabletNum;
-                                    dbReplicaNum += tablet.getReplicas().size();
+                                    ++tableNum;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/StatisticProcDir.java:150]:
```suggestion
                                    ++tabletNum;
```

**Comment by morningman** [comment]
Hi @ccoffline , please rebase the master to solve the conflict

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/StatisticProcDir.java (hunk 1 above)
Comment (ccoffline) [reply fe/fe-core/src/main/java/org/apache/doris/common/proc/StatisticProcDir.java:137]: ok

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**